### ISSUE TITLE
Better optimization options

### ISSUE THREAD

**Issue by wpm** [wpm/mycroft#40]
1. Add learning rate as a parameter since it is common to all optimizers.
2. Optionally use a [reduce learning rate on plateau](https://keras.io/callbacks/#reducelronplateau) callback.
3. Optionally use an [early stopping](https://keras.io/callbacks/#earlystopping) callback.

These may help eke out a few more good epochs from training.
